### PR TITLE
Remove kubecfg, expose kubectl in its place

### PR DESCRIPTION
**Pull request by smarterclayton**
Now that kubectl upstream has deprecated kubecfg, do the same for
OpenShift.  This updates `openshift kube` to point to kubectl, exposed
directly as-is from upstream. It includes the new resize, run-container,
and rollingupdate commands.

Clean up and reorganize the "experimental" commands option to be a new,
visible sub-command `openshift ex`, which has tokens and config under
it for now.  Remove the openshift-experimental binary.

When printing help directly, use os.Stdout.

Add more tests to hack/test-cmd.sh, and print slightly more logging.

@mfojtik @fabianofranz review please

### CODE REVIEW

**Comment by smarterclayton** [comment]
I brought back kubectl as its own command, because it prevents us from having to expose rolling update, resize, and run-container (which we would want to apply to deployment configs) directly.

File: pkg/cmd/cli/cli.go [hunk 1]
```diff
@@ -76,6 +75,23 @@ func NewCommandCLI(name string) *cobra.Command {
 	return cmds
 }
 
+// NewCmdKubectl provides exactly the functionality from Kubernetes,
+// but with support for OpenShift resources
+func NewCmdKubectl(name string) *cobra.Command {
+	flags := pflag.NewFlagSet("", pflag.ContinueOnError)
+	clientcmd.DefaultCluster.Server = "https://localhost:8443"
```
Comment (mfojtik) [inline pkg/cmd/cli/cli.go]: shouldn't this be a constant?

File: pkg/cmd/cli/cmd/factory.go [hunk 2]
```diff
@@ -96,3 +96,29 @@ func (f *Factory) Clients(cmd *cobra.Command) (*client.Client, *kclient.Client,
...
+	return e.RESTMapper.VersionAndKindForResource(resource)
+}
+
+// expandResourceShortcut will return the expanded version of resource
+// (something that a pkg/api/meta.RESTMapper can understand), if it is
+// indeed a shortcut. Otherwise, will return resource unmodified.
+func expandResourceShortcut(resource string) string {
+	shortForms := map[string]string{
+		"dc": "deploymentConfigs",
+		"bc": "buildConfigs",
```
Comment (mfojtik) [inline pkg/cmd/cli/cmd/factory.go]: can you add one more? 'bl' (buildLogs)

**Comment by mfojtik** [comment]
LGTM

File: hack/test-cmd.sh [hunk 3]
```diff
@@ -91,6 +95,13 @@ export OPENSHIFT_PROFILE="${CLI_PROFILE-}"
 # Begin tests
 #
 
+# verify some default commands
+[ "$(openshift ex)" ]
+[ "$(openshift ex config 2>&1)" ]
+[ "$(openshift ex tokens)" ]
+[ "$(openshift kubectl)" ]
+[ "$(openshift kube 2>&1)" ]
```
Comment (fabianofranz) [inline hack/test-cmd.sh]: [ "$(openshift cli)" ]

**Comment by fabianofranz** [comment]
Nice PR, LGTM

**Comment by smarterclayton** [comment]
Issues fixed, [merge]

**Pull request merged**